cypress/e2e/api.cy.js

```diff
@@ -0,0 +1,162 @@
+describe('EngageSphere', () => {
+  it('returns the correct status and body structure', () => {
+    cy.api({
+      method: 'GET',
+      url: `${Cypress.env('apiUrl')}/customers`
+    }).then(({ status, body }) => {
+      expect(status).to.eq(200)
+      expect(body).to.have.property('customers')
+
+      expect(body.customers[0]).to.have.property('id')
+      expect(body.customers[0]).to.have.property('name')
+      expect(body.customers[0]).to.have.property('employees')
+      expect(body.customers[0]).to.have.property('industry')
+      expect(body.customers[0]).to.have.property('contactInfo')
+      expect(body.customers[0]).to.have.property('address')
+      expect(body.customers[0]).to.have.property('size')
+
+      if (body.customers[0].contactInfo) {
+        expect(body.customers[0].contactInfo).to.have.property('name')
+        expect(body.customers[0].contactInfo).to.have.property('email')
+      }
+
+      if (body.customers[0].address) {
+        expect(body.customers[0].address).to.have.property('street')
+        expect(body.customers[0].address).to.have.property('city')
+        expect(body.customers[0].address).to.have.property('state')
+        expect(body.customers[0].address).to.have.property('zipCode')
+        expect(body.customers[0].address).to.have.property('country')
+      }
+
+      expect(body).to.have.property('pageInfo')
+      expect(body.pageInfo).to.have.property('currentPage')
```

Comment:
E aqui poderia ser:

`expect(body.pageInfo).to.have.all.keys('currentPage', 'totalPages', 'totalCustomers')`